lib/components/ms-code/MsCodeValidation.vue

```diff
@@ -0,0 +1,249 @@
+<template>
+  <div class="code-input">
+    <div class="code-input-list">
+      <template
+        v-for="i in codeLength || 6"
+        :key="`code${i - 1}`"
+      >
+        <ion-input
+          ref="inputs"
+          v-model="codes[i - 1]"
+          class="code-input-list__item title-h1-xl"
+          :class="{ 'has-values': codes[i - 1] !== '' }"
+          type="text"
+          inputmode="numeric"
+          :maxlength="1"
+          @keydown="onKeydown($event)"
+          @ion-input="onIonInput($event)"
+          @keyup.backspace="onKeyUpBackspace($event)"
+          @paste="onPaste($event)"
+        />
+      </template>
+    </div>
+    <div class="code-input-message">
+      <ion-text
+        class="code-input-message__invalid subtitle-sm"
+        v-if="isValid === false"
+      >
+        {{ $msTranslate('lib.components.msCodeValidationModal.codeNotValid') }}
+      </ion-text>
+      <ion-text
+        class="code-input-message__valid subtitle-sm"
+        v-if="isValid === true"
+      >
+        {{ $msTranslate('lib.components.msCodeValidationModal.codeValid') }}
+      </ion-text>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { nextTick, onMounted, ref, defineEmits } from 'vue';
+import { IonInput, IonText } from '@ionic/vue';
+import { Ref } from 'vue';
+const inputs = ref();
+const codes = ref<string[]>([]);
+const isValid: Ref<undefined | boolean> = ref(undefined);
```

Comment:
We should separate two notions:
- code `correctness`, meaning is it the right code (does it match?)
- code `validity`, meaning does it have the required amount of digits?

It's a bit confusing right now.